source/collections/Queue.ooc

```diff
@@ -0,0 +1,51 @@
+
+Queue: class <T> {
+	_queue: T*
+	_capacity: Int
+	capacity ::= this _capacity
+	_count: Int
+	count ::= this _count
+	_head: Int
+	_tail: Int
+
+	init: func (=_capacity) {
+		this _queue = gc_calloc(capacity as SizeT, T size)
+		this _head = 0
+		this _tail = 0
+	}
+
+	free: override func {
+		gc_free(_queue)
+		super()
+	}
+
+	isEmpty: func -> Bool {
+		this _count == 0
+	}
+
+	isFull: func -> Bool {
+		this _count == this _capacity
+	}
+
+	add: func (item: T) {
```

Comment:
These are the add/remove function pairs I've seen for queues in other programming languages:
- C#: enqueue/dequeue
- C++: push/pop
- Java: add/remove
... and I think we shouldn't mix the pairs. The original, C# version of Kean uses the C# names. @simonmika?